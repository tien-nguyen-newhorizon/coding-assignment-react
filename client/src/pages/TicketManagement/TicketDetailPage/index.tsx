import SuspenseWrapper from '@client/components/SuspenseWrapper';
import TicketDetail from '@client/modules/TicketManagement/TicketDetail';

const TicketDetailPage = () => {
  return (
    <SuspenseWrapper>
      <TicketDetail />
    </SuspenseWrapper>
  );
};

export default TicketDetailPage;
