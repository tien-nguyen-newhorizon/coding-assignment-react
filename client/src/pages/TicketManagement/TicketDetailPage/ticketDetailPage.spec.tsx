import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import TicketDetailPage from './index';

// 1. Mock the dependencies
jest.mock('@client/components/SuspenseWrapper', () => {
  return ({ children }: { children: React.ReactNode }) => <div data-testid="mock-suspense-wrapper">{children}</div>;
});

jest.mock('@client/modules/TicketManagement/TicketDetail', () => {
  return () => <div data-testid="mock-ticket-detail">Mock Ticket Detail Content</div>;
});

describe('TicketDetailPage', () => {
  // --- Render Test ---
  it('should render the SuspenseWrapper and TicketDetail components', () => {
    render(<TicketDetailPage />);

    const suspenseWrapper = screen.getByTestId('mock-suspense-wrapper');
    expect(suspenseWrapper).toBeInTheDocument();

    const ticketDetail = screen.getByTestId('mock-ticket-detail');
    expect(ticketDetail).toBeInTheDocument();
    expect(suspenseWrapper).toContainElement(ticketDetail);

    expect(screen.getByText('Mock Ticket Detail Content')).toBeInTheDocument();
  });

  // --- Snapshot Test ---
  it('should match the snapshot', () => {
    const { asFragment } = render(<TicketDetailPage />);
    expect(asFragment()).toMatchSnapshot();
  });
});
