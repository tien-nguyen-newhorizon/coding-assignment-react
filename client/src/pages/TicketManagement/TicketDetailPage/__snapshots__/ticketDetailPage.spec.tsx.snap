// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TicketDetailPage should match the snapshot 1`] = `
<DocumentFragment>
  <div
    data-testid="mock-suspense-wrapper"
  >
    <div
      data-testid="mock-ticket-detail"
    >
      Mock Ticket Detail Content
    </div>
  </div>
</DocumentFragment>
`;
